apps/openmw/mwworld/projectilemanager.cpp

```diff
@@ -501,27 +529,153 @@ namespace MWWorld
                     caster = result.mHitObject;
 
                 MWMechanics::projectileHit(caster, result.mHitObject, bow, projectileRef.getPtr(), result.mHit ? result.mHitPos : newPos, it->mAttackStrength);
+                if (result.mProjectileId >= 0)
+                    markToRemove(result.mProjectileId);
 
                 if (underwater)
                     mRendering->emitWaterRipple(newPos);
 
-                mParent->removeChild(it->mNode);
+                cleanupProjectile(*it);
                 it = mProjectiles.erase(it);
+
                 continue;
             }
 
             ++it;
         }
     }
 
+    void ProjectileManager::manualHit(int projectileId, const MWWorld::Ptr& target, const osg::Vec3f& pos)
+    {
+        for (std::vector<ProjectileState>::iterator it = mProjectiles.begin(); it != mProjectiles.end(); ++it)
+        {
+            if (it->mProjectileId == projectileId)
+            {
+                if (it->mHit)
+                    return;
+
+                MWWorld::Ptr caster = it->getCaster();
+                if (caster == target)
+                    return;
+
+                if (!isValidTarget(caster, target))
+                    return;
+
+                if (caster.isEmpty())
+                    caster = target;
```

Comment:
Not sure I understand why the case of empty caster is not a logic error? There is another check for empty below that seems redundant now.

Reply:
I can not really tell - too many time passed, but I am pretty sure that was check was added to do not break Morrowind's game mechanics.

Reply:
As I understand this is a copy of logic [here](https://github.com/OpenMW/openmw/pull/2183/files#diff-cc9df71b3061f0d452a113f4708feb57R528-R529) that was originally introduced by [this](https://github.com/OpenMW/openmw/commit/12de0afb03a382332f2da3d31b073fe9d49bb7ff#diff-d3b808d6f40e4c12e247457971def846R2294-R2295) commit. Looks like some kind of a workaround for an undescribed problem. And of course there is no trivial way to find a pr for the original commit to see the discussion.

Reply:
Found the [pr](https://github.com/zinnschlag/openmw/pull/1279/files#diff-d3b808d6f40e4c12e247457971def846R2299-R2300). Had to look manually all pr around commit date. Sadly there is no any discussion.